### PR TITLE
Introduce Special SummaryState Handling for Region Variables

### PR DESCRIPTION
**Pull request by bska**
This is in preparation of adding support for using region-level summary vectors in the defining expressions of field-level UDQs.

### CODE REVIEW

**Comment by bska** [comment]
I'm creating this PR in draft mode because it is part of a PR chain building up to enabling using region-level summary vectors in the defining expressions of field-level UDQs.  I will keep the PR up-to-date with development elsewhere, and in a draft state, until such time as it is ready for review.

**Comment by bska** [comment]
The upstream PR was merged into the master branch.  I'm marking this PR as "ready for review" now and I'm running a build check.

**Comment by bska** [comment]
jenkins build this please

File: opm/input/eclipse/Schedule/SummaryState.cpp [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 #include <opm/input/eclipse/Schedule/UDQ/UDQSet.hpp>
 
+#include <opm/io/eclipse/SummaryNode.hpp>
```
Comment (akva2) [inline opm/input/eclipse/Schedule/SummaryState.cpp:24]: i < c

**Review by akva2 — APPROVED** [review]

File: opm/input/eclipse/Schedule/SummaryState.cpp (hunk 1 above)
Comment (bska) [reply opm/input/eclipse/Schedule/SummaryState.cpp:26]: > i < c ?

I've pushed an update to address this.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
PR approved and build check is green.  I'll merge into master.

**Pull request merged**